pages/common/fakedata.md

```diff
@@ -0,0 +1,24 @@
+# fakedata
+
+> Generate fake data using a large variety of generators.
+> More information: <https://github.com/lucapette/fakedata>.
+
+- List all valid generators:
+
+`fakedata --generators`
+
+- Generate data using one or more generators:
+
+`fakedata {{generator1}} {{generator2}}`
+
+- Generate data with a specific output format:
+
+`fakedata --format {{csv|tab|sql}} {{generator}}`
+
+- Generate {{n}} data:
+
+`fakedata --limit {{n}} {{generator}}`
+
+- Generate data using a custom output template (capitalize the first letter of generators):
+
+`echo "value: {{Generator}}" | fakedata`
```

Comment:
```suggestion
`echo "value: {{generator}}" | fakedata`
```

Reply:
I used "Camel case" on purpose because:
> If the generator name is a single word, then it's available as a function with the same name capitalized (example: int becomes Int). If the generator name is composed by multiple words joined by dots, then the function name is again capitalized by the first letter of the word and joined together (example: product.name becomes Product.Name).

See https://github.com/lucapette/fakedata#generators-1

What do you think of this "intentional violation"? 😇

Reply:
:thinking: all the custom format is the custom argument option in this example, so the valid highlight is the following:
```suggestion
`echo "{{value: Generator}}" | fakedata`
```

Reply:
I agree with it.  It's required, so definitely keep it.  @navarroaxel see the example description itself; it mentions this oddity.

Reply:
@nicokosi maybe it would be better to change the parenthetic message to `the first letter of generator names must be capitalized`, that sounds a bit clearer.

Reply:
Done in commit 7373486e.

Reply:
> @nicokosi maybe it would be better to change the parenthetic message to `the first letter of generator names must be capitalized`, that sounds a bit clearer.

Done in commit 7373486e.

Reply:
@nicokosi I believe @navarroaxel is right about the extents of the token syntax; "value: " is your own custom text and there fore should be highlighted as needed. https://github.com/tldr-pages/tldr/pull/5675/files#r606702953

Reply:
Actually, I see there's an issue here.  `fakedata` has a special syntax for templates where you enclose the capitalized template name in `{{` and `}}`.  Both the capitalization and the braces are required because `fakedata` uses the [Go `template` library](https://golang.org/pkg/text/template/).  The issue with that is that we at tldr-pages use that very syntax for a different purpose: that of highlighting user-entered values.

Instead of the original text here and @navarroaxel's suggestion, the appropriate would be this:
```
echo "{{value: {{Generator}}}}" | fakedata`
```
However, many clients do a blind replace of `{{` and `}}` with HTML/ANSI escape codes that highlight the text, so we might need some sort of escaping, e.g.
```
echo "{{value: \{\{Generator\}\}}}" | fakedata`
```
But I'm just not sure.  @sbrl what should we do here?

Reply:
> Actually, I see that there's an issue here. `fakedata` has a special syntax for their templates where you enclose the capitalized template name in `{{` and `}}`. Both the capitalization and the braces are required because `fakedata` uses the [Go `template` library](https://golang.org/pkg/text/template/). The issue with that is that we at tldr-pages use that very syntax for a different purpose: that of highlighting user-entered values.
> 
> Instead of the original text here and @navarroaxel's suggestion, the appropriate would be this:
> 
> ```
> echo "{{value: {{Generator}}}}" | fakedata`
> ```
> 
> However, many clients do a blind replace of `{{` and `}}` with HTML/ANSI escape codes that highlight the text, so we might need some sort of escaping, e.g.
> 
> ```
> echo "{{value: \{\{Generator\}\}}}" | fakedata`
> ```
> 

Done in 8ce90227.